lib/rules/lint-no-curly-component-invocation.js

```diff
@@ -80,7 +80,7 @@ module.exports = class NoCurlyComponentInvocation extends Rule {
 
   getCurlyInvocationSyntaxError(node) {
     let name = node.path.original;
-    if (node.path.type !== 'PathExpression') {
+    if (node.path.type !== 'PathExpression' || this.isLocal(node)) {
```

Comment:
```suggestion
    let hasArguments = node.params.length > 0 || node.hash.pairs.length > 0;
    let isAmbiguousLocalInvocation = this.isLocal(node) && !hasArguments;
    if (node.path.type !== 'PathExpression' || isAmbiguousLocalInvocation) {
```

Reply:
I've updated the logic based on your feedback but needed to tweak it slightly. I had to move calculating `isAmbiguousLocalInvocation` check to happen lazily because the `this.isLocal` throws an exception when the type in the mustache is not a `PathExpression`, e.g. `NumberLiteral.

So can `{{whatever some=arg}}` or `{{whatever arg}}` ever be a helper or just a component?